### DISCUSSION TITLE
Expose some Immutable

### DISCUSSION THREAD

**Discussion by workgroupengineering** [AvaloniaUI/Avalonia#13178] [Ideas]
if I need to create `IImmutableBrush` I have to do:

```csharp
new ImageBrush(bitmap)
        {
            AlignmentX = AlignmentX.Left,
            AlignmentY = AlignmentY.Top,
            Stretch = Stretch.Fill
        }.ToImmutable();
```

I think it's a waste to create the mutable version when I already know I need an immutable one.

We could make the constructors public or better yet create a factory like `Immutable.Create(...`

**Comment by maxkatz6** [comment]
Most of immutable brushes are already public. Immutable image brush though (and possible some other tile brush variants) were added recently with composition renderer and were planned to be more deeply integrated at some point into rendering pipeline.
